CRM/Sqltasks/Action/CreateActivity.php

```diff
@@ -50,6 +51,18 @@ public function getDefaultOrder() {
    */
   public function execute() {
     $this->resetHasExecuted();
+
+    if ($this->getConfigValue('store_activity_ids')) {
+      $contact_table = $this->getContactTable();
+
+      if (strpos($contact_table, 'civicrm_') === 0) {
+        throw new CRM_Core_Exception("Cannot alter table $contact_table");
+      }
+
+      $this->contact_table_ai_col = self::addAutoIncrementColumn($contact_table);
+      CRM_Core_DAO::executeQuery("ALTER TABLE `$contact_table` ADD `sqltask_activity_id` INT");
```

Comment:
Let's add a sanity check here in case someone uses this action twice on the same table: First check if the column exists (`SHOW COLUMNS FROM $contact_table LIKE 'sqltask_activity_id'`).

If it exists, add a small warning (something like `Overwriting existing values for sqltask_activity_id in $contact_table`) to the task log and skip the `ALTER TABLE`.